scripts/mysql_system_tables_fix.sql

```diff
@@ -171,6 +171,14 @@ ALTER TABLE user
 ALTER TABLE user
   ADD Password char(41) character set latin1 collate latin1_bin NOT NULL default '' AFTER User;
 
+# In MySQL the Unix socket authentication plugin has a different name. Thus the
+# references to if need to be renamed in the user table. Thanks to the WHERE
+# clauses this applies only to MySQL->MariaDB upgrades and nothing else.
+UPDATE user
+  SET plugin='unix_socket' WHERE plugin='auth_socket';
+UPDATE plugin
+  SET name='unix_socket', dl='unix_socket.so' WHERE name='auth_socket';
```

Comment:
The issue with updating the `mysql.plugin` table here is that in `10.4+`, `unix_socket` is installed by default, thus the entry shouldn't be in `mysql.plugin`. The `UPDATE` here should rather be a `DELETE`.
But in `10.3`, the row should be there after the upgrade, so the server knows it needs to load the plugin at startup. One issue in this version is that the plugin is named `unix_socket`, but the shared library is named `auth_socket.so`.

Reply:
We want to ensure there are no references to `auth_socket` anywhere. Therefore we do two renames here. At least for users the rename is essential. For plugins the rename does not hurt. I will check if a DELETE would also be enough but renaming is at least the safest option.

Reply:
PR updated as requested by @robertbindar - I also tested that this works as well.